Memory/src/main/scala/com/thoughtworks/compute/Memory.scala

```diff
@@ -261,6 +261,33 @@ object Memory extends LowPriorityMemory {
 
   // TODO: short, bool, char
 
+  implicit object CharMemory extends NioMemory[Char] {
+
+    override type HostBuffer = CharBuffer
+
+    override def fromByteBuffer(byteBuffer: ByteBuffer): CharBuffer = byteBuffer.asCharBuffer()
+
+    override def numberOfBytesPerElement: Int = java.lang.Character.BYTES
+
+    override def address(buffer: CharBuffer): Long = MemoryUtil.memAddress(buffer)
+
+    override def get(buffer: CharBuffer, index: Int): Char = buffer.get(index)
+
+    override def put(buffer: CharBuffer, index: Int, value: Char): Unit = buffer.put(index, value)
+
+    override def allocate(numberOfElement: Int): CharBuffer = MemoryUtil.memAlloc(2 * numberOfElement).asCharBuffer()
```

Comment:
It's better to avoid the hard-coded number `2`

Reply:
Yes, indeed.